smpclient/__init__.py

```diff
@@ -154,6 +157,85 @@ async def upload(
                 logger.error(message)
                 raise SMPUploadError(message)
 
+    async def upload_file(
+        self,
+        file_data: bytes,
+        file_path: str,
+        first_timeout_s: float = 40.000,
+        subsequent_timeout_s: float = 2.500,
+    ) -> AsyncIterator[int]:
+        response = await self.request(
+            self._maximize_file_upload_packet(
+                FileUpload(name=file_path, off=0, data=b"", len=len(file_data)),
+                file_data,
+            ),
+            timeout_s=first_timeout_s,
+        )
```

Comment:
The `first_timeout_s` and `subsequent_timeout_s` parameters were informed by MCUmgr for FW uploads where the user has not configured "erase flash progressively" or "stream flash" and the device may take a while to do an erase on all the pages at once.

Let's start by assuming we don't need to do anything funny like that and just set one timeout: timeout_s = 2.500